src/resources/Catalogs/tests/Catalog.spec.ts

```diff
@@ -0,0 +1,80 @@
+import API from '../../../APICore';
+import Catalog from '../Catalog';
+import {CatalogModel, CatalogsListOptions, NewCatalogModel} from '../Interfaces';
+
+jest.mock('../../../APICore');
+
+const APIMock: jest.Mock<API> = API as any;
```

Comment:
Is the cast here necessary? You seem to cast back to `jest.Mocked<API>` further down

Reply:
Yes if I don't cast here, then typescript won't allow me to create an `API` without params. ([line 11](https://github.com/coveo/platform-client/pull/9/files/8bd363d99efafa681bf4dd0c48a13dd9b355d477#diff-0e5287af84ddaf2ec60933a88a32fe2bR11))